eng/ci/templates/official/jobs/run-benchmarks.yml

```diff
@@ -77,11 +83,35 @@ jobs:
       arguments: -c Release -o $(functionAppOutputPath) -f net9.0
       workingDirectory: $(Build.ArtifactStagingDirectory)/PerformanceTestApps/$(functionApp)
 
-  - script: dotnet tool install -g Microsoft.Crank.Controller --version "0.2.0-*"
-    displayName: Install Microsoft.Crank.Controller
+  - task: AzureCLI@2
+    displayName: Get Remote Agent IP Address
+    inputs:
+      azureSubscription: $(ServiceConnection)
+      scriptType: pscore
+      scriptLocation: inlineScript
+      inlineScript: |
+        do {
+          $Entity = az storage entity show `
+            --account-name $(StorageAccount) `
+            --auth-mode login `
+            --table-name $(BenchmarkAgentInfoTableName) `
+            --partition-key $(Build.BuildNumber) `
+            --row-key $(agentName) `
+            --select AgentIPAddress
+
+          if ($Entity) {
+            $AgentIPAddress = ($Entity | ConvertFrom-Json).AgentIPAddress
+            Write-Host "##vso[task.setvariable variable=appAgentIpAddress]$AgentIPAddress"
+            break
+          } else {
+            Write-Host "Entity not found. Retrying in 30 seconds..."
```

Comment:
may want to add a timeout and failure to this step so it doesn't spin for an hour.

Reply:
Good call. Added a retry exit mechanism after 30 minutes.